01-messaging.md

```diff
@@ -153,6 +156,73 @@ diagnosis, as it indicates that one peer has a bug.
 It may be wise not to distinguish errors in production settings, lest
 it leak information, thus the optional data field.
 
+## Control Messages
+
+### The `ping` message
+
+In order to allow for the existence of very long-lived TCP connections, at
+times it may be required that both ends keep alive the TCP connection at the
+application level.
+
+1. type: 18 (`ping`)
+2. data: 
+    * [2:num_pong_bytes]
+
+#### Requirements
+
+A node receiving a `ping` message MUST respond with a `pong` message that
+contains the specified `num_pong_bytes` number of bytes within it's data
+payload. These `ping` messages may be sent any time at the sender's discretion,
+however `ping` messages MUST NOT be sent with a frequency lower than one every
+30 seconds.
+
```

Comment:
Usually clearest to separate sender and receiver requirements (eg. what happens if the sender violates spec and flood pings?).  We should also specify padding on send is allowed, and MBZ for future extensions.

The sender MAY send `ping` messages MAY be sent any time at the sender's discretion, but MUST NOT send `ping` more often than than once every 30 seconds, and MUST NOT set `num-pong-bytes` greater than 65533.  The sender MAY pad the `ping` message after the `num-pong-bytes` field with zeroes, and MUST NOT use non-zero padding.

The receiver SHOULD fail the connection if `num-pong-bytes` is greater than 65535.